app/CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+# helpful cmake package for custom macros
+include(CMakeParseArguments)
+
+#################################################################################################
+# CMake macro 'declare_program'
+# Declare a new program
+# Arguments:
+#   name - name of program, same as name of source file but without .cpp extension
+#   dependencies - CMake target dependencies
+#################################################################################################
+macro(declare_program)
+
+  #set(options ) # options are parameters with no values
+  set(oneValueArgs name) #single value parameters
+  set(multiValueArgs dependencies) #multiple value parameters
+  cmake_parse_arguments(declare_program "${options}" "${oneValueArgs}"
+                        "${multiValueArgs}" ${ARGN})
+
+  add_executable(${declare_program_name} ${declare_program_name}.cpp)
+  #deal_ii_setup_target(${declare_program_name})
+  target_link_libraries(${declare_program_name} PRIVATE ${declare_program_dependencies})
+  install(TARGETS ${declare_program_name} DESTINATION bin)
+endmacro()
```

Comment:
This is a very simple CMake macro for defining a new program, linking it to other CMake targets, and the installing it to the `bin` subdirectory of `CMAKE_INSTALL_PREFIX`. I defined this just to save time in development and it is sloppy because it makes several assumptions.
- the executable name is the same as the source file name (without the `.cpp` extension)
- the executable source file is in the same directory as the CMakeLists.txt file this is called from
- the executable source file is a cpp file
These assumptions very quickly break down as you add your CUDA executables, but this macro does give you the template commands for defining a new CMake executable target.